pkg/cmd/cli/nodeagent/server.go

```diff
@@ -333,6 +337,72 @@ func (s *nodeAgentServer) markInProgressCRsFailed() {
 	s.markInProgressPVRsFailed(client)
 }
 
+func (s *nodeAgentServer) markDataUploadsCancel(r *controller.DataUploadReconciler) {
+	// the function is called before starting the controller manager, the embedded client isn't ready to use, so create a new one here
+	client, err := ctrlclient.New(s.mgr.GetConfig(), ctrlclient.Options{Scheme: s.mgr.GetScheme()})
+	if err != nil {
+		s.logger.WithError(errors.WithStack(err)).Error("failed to create client")
+		return
+	}
+	if dataUploads, err := r.FindDataUploads(s.ctx, client, s.namespace); err != nil {
+		s.logger.WithError(errors.WithStack(err)).Error("failed to find data downloads")
+	} else {
+		for i := range dataUploads {
+			du := dataUploads[i]
+			if du.Status.Phase == velerov2alpha1api.DataUploadPhaseAccepted ||
```

Comment:
Suggest to add a wrapper function to do this cancellation centrally, this function will:
1. Judge if the phase is cancellation
2. Set `Cancel` flag
3. Set any other fields for terminal status, i.e., `StartTimestamp`, `CompletionTimestamp`
4. Patch the DR
5. ...

Or in another words, move this if block to the wrapper function.

As you can see, the Velero server and node-agent shares the same block of code. Moreover, this function could also be used by other scenarios, i.e., backup deletion

Reply:
As we discussed, keep the code unchanged until backup deletion really needs one wrapper function